Update instructions for GitHub API tokens (#406)
Mic92/nixpkgs-review: README.md

`/etc/nix/machines` or via the `nix.buildMachines` nixos options in
`configuration.nix`. This allows to parallelize builds across multiple machines.

## GitHub API token

Some commands (i.e., `post-result` or `merge`) require a GitHub API token, and
even for read-only calls, GitHub returns 403 error messages if your IP hits the
rate limit for unauthenticated calls.

> Nixpkgs-review will automatically read the oauth_token stored by [hub][1] or
> [gh][2] if they are installed.

Reviewer: ```suggestion
Nixpkgs-review will automatically read the oauth_token stored by [hub](https://hub.github.com/) or
[gh](https://cli.github.com/) if they are installed.
```

> Otherwise, you'll have to manually create a "personal access token (classic)"
> through the GitHub website. See [the GitHub documentation][3] for

Reviewer: ```suggestion
Otherwise, you'll have to create a "personal access token (classic)"
through GitHub's website. See [the GitHub documentation](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#personal-access-tokens-classic) for
```

> instructions. If you plan to post comments with the reports generated, make sure

Reviewer: ```suggestion
instructions. If you plan to post the generated reports, make sure
```

> to give `public_repo` scope to the token.

Reviewer: ```suggestion
to give it the `public_repo` scope.
```

Then use either the `GITHUB_TOKEN` environment variable or the `--token`
parameter of the `pr` subcommand to supply your token to nixpkgs-review.

```console
$ GITHUB_TOKEN=ghp_WAI7vpi9wVHbxPOA185NwWvaMawDuCnMGc3E nixpkgs-review pr 37244 --post-result
$ nixpkgs-review pr 37244 --token ghp_WAI7vpi9wVHbxPOA185NwWvaMawDuCnMGc3E --post-result
```

> [1]: https://hub.github.com/
> [2]: https://cli.github.com/
> [3]: https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#personal-access-tokens-classic
> 

Reviewer: ```suggestion
```
